optimade/server/config.py

```diff
@@ -137,42 +138,47 @@ def set_implementation_version(cls, v):
         res.update(v)
         return res
 
-    @validator("log_level")
-    def force_debug_log_level(cls, v, values):
-        """If `debug` is `True`, then force `log_level` to be `DEBUG` as well"""
-        from optimade.server.logger import CONSOLE_HANDLER
-
-        if values.get("debug", False):
-            v = LogLevel.DEBUG
-        CONSOLE_HANDLER.setLevel(v.value.upper())
-        return v
-
     @root_validator(pre=True)
-    def load_default_settings(cls, values):  # pylint: disable=no-self-argument
+    def load_settings(cls, values):  # pylint: disable=no-self-argument
         """
-        Loads settings from a root file if available and uses that as defaults in
-        place of built in defaults
+        Loads settings from a JSON config file, if available, and uses them in place
+        of the built-in defaults.
         """
-        from optimade.server.logger import LOGGER
-
         config_file_path = Path(values.get("config_file", DEFAULT_CONFIG_FILE_PATH))
 
         new_values = {}
 
-        if config_file_path.exists() and config_file_path.is_file():
-            LOGGER.debug("Found config file at: %s", config_file_path)
-            with open(config_file_path) as f:
-                new_values = json.load(f)
+        if config_file_path.is_file():
+            try:
+                with open(config_file_path) as f:
+                    new_values = json.load(f)
+            except json.JSONDecodeError as exc:
+                raise RuntimeError(f"Error in JSON file {config_file_path}: {exc}")
+
+        elif DEFAULT_CONFIG_FILE_PATH != str(config_file_path):
+            raise RuntimeError(
+                f"Unable to find requested config file at {config_file_path}"
+            )
+
         else:
-            LOGGER.debug(  # pragma: no cover
-                "Did not find config file at: %s", config_file_path
+            warnings.warn(
+                f"Unable to find config file in default location {DEFAULT_CONFIG_FILE_PATH}, "
+                "using the built-in default settings."
```

Comment:
So originally, I didn't want to do warnings here, but instead logged warnings due to warnings being part of the response now. I.e., you're essentially responding with server warnings in a place that is meant to show warnings related to something OPTIMADE API-specific.

Reply:
Could we instead retrieve the `uvicorn` logger and log to that one?
The only potential error I see for this is if one chooses to not use `uvicorn`...

Reply:
I was worried about this too, but it doesn't seem like the warning gets picked up by the MiddleWare as its the config is loaded before the app is created. I think your other suggestion about logging the config file when (maybe just before) the server itself starts is probably best; I was thinking of making the `config_file` setting optional (which it is, as you could use Python/env vars to set it up), then if `config_file` is None just log a warning about defaults.

Reply:
> I was worried about this too, but it doesn't seem like the warning gets picked up by the MiddleWare as its the config is loaded before the app is created. (...)

Right! This might have been my original motivation for the "lazy importing" that I changed after a review by @shyamd.
Granted, he only pointed out two places, I then chose to move all import to the top-level... But then again, it would still need to load the CONFIG before finishing handling the middleware... So probably it won't work anyway 😅

Reply:
> (...) I was thinking of making the `config_file` setting optional (which it is, as you could use Python/env vars to set it up), then if `config_file` is None just log a warning about defaults.

Yeah, I think this might be the best option, honestly. I thought about doing this for the log-related configuration options as well. But it seems reasonable to have that one _only_ be customizable through an environment variable.